.github/workflows/commit-linter.yaml

```diff
@@ -0,0 +1,38 @@
+name: commit-linter
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  commit-lint:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: cachix/install-nix-action@v25
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+
+      - uses: DeterminateSystems/magic-nix-cache-action@main
+
+      - name: npm install
+        run: nix develop --command npm install --include=dev
+
+      - name: Print versions
+        run: |
+          git --version
+          node --version
+          npm --version
+          npx commitlint --version
+
+      - name: Validate current commit (last commit) with commitlint
+        if: github.event_name == 'push'
+        run: nix develop --command npx commitlint --last
+
+      - name: Validate PR commits with commitlint
+        if: github.event_name == 'pull_request'
+        run: nix develop --command npx commitlint --from ${{ github.event.pull_request.head.sha }}~${{ github.event.pull_request.commits }} --to ${{ github.event.pull_request.head.sha }}
```

Comment:
@KingMob This all looks correct. One modification might be to add `-V` flag to `commitlint` so we  can introspect what was checked  as recommended in your update to the README
```
$  npx commitlint -f head~ -t head -V
⧗   input: feat: Add Conventional Commits linter for CI
✔   found 0 problems, 0 warnings
```

Reply:
OK, will add